From a9950cedd71dcc47af5b2c722e43faac610f18c5 Mon Sep 17 00:00:00 2001
From: Tomasz Maciej Nowak <tomek_n@o2.pl>
Date: Wed, 24 Jan 2018 16:41:32 +0100
Subject: [PATCH 7/9] mvebu: base-files: moving on from net-tools

Remove dependency on net-tools, replacing calls to ifconfig with iproute
equivalent, in functions changing interface MAC addresses.

Signed-off-by: Tomasz Maciej Nowak <tomek_n@o2.pl>
---
 target/linux/mvebu/base-files/lib/preinit/06_set_iface_mac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/target/linux/mvebu/base-files/lib/preinit/06_set_iface_mac b/target/linux/mvebu/base-files/lib/preinit/06_set_iface_mac
index e884351bf7..b56cbf50ad 100644
--- a/target/linux/mvebu/base-files/lib/preinit/06_set_iface_mac
+++ b/target/linux/mvebu/base-files/lib/preinit/06_set_iface_mac
@@ -11,8 +11,8 @@ preinit_set_mac_address() {
 	case $(board_name) in
 	armada-xp-linksys-mamba)
 		mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
-		ifconfig eth0 hw ether $mac 2>/dev/null
-		ifconfig eth1 hw ether $mac 2>/dev/null
+		ip link set dev eth0 address $mac 2>/dev/null
+		ip link set dev eth1 address $mac 2>/dev/null
 		;;
 	armada-385-linksys-caiman|armada-385-linksys-cobra|armada-385-linksys-rango|armada-385-linksys-shelby)
 		# rename interfaces back to the way they were with 4.4
@@ -26,8 +26,8 @@ preinit_set_mac_address() {
 
 		mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
 		mac_wan=$(macaddr_setbit_la "$mac")
-		ifconfig eth1 hw ether $mac 2>/dev/null
-		ifconfig eth0 hw ether $mac_wan 2>/dev/null
+		ip link set dev eth1 address $mac 2>/dev/null
+		ip link set dev eth0 address $mac_wan 2>/dev/null
 		;;
 	armada-385-db-ap|armada-388-clearfog)
 		# rename interfaces back to the way they were with 4.4
-- 
2.16.2

